api/krusty/remoteload_test.go

```diff
@@ -41,103 +28,254 @@ spec:
   containers:
   - image: nginx:1.7.9
     name: nginx
-`, string(yml))
+`
+
+type remoteResourceCase struct {
+	skip          bool
+	kustomization string
 }
 
-func TestRemoteResource(t *testing.T) {
+func createKusDir(content string, assert *assert.Assertions) (filesys.FileSystem, filesys.ConfirmedDir) {
 	fSys := filesys.MakeFsOnDisk()
-	b := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
 	tmpDir, err := filesys.NewTmpConfirmedDir()
-	assert.NoError(t, err)
-	assert.NoError(t, fSys.WriteFile(filepath.Join(tmpDir.String(), "kustomization.yaml"), []byte(`
-resources:
-- github.com/kubernetes-sigs/kustomize/examples/multibases/dev/?ref=v1.0.6
-`)))
+	assert.NoError(err)
+	assert.NoError(fSys.WriteFile(filepath.Join(tmpDir.String(), "kustomization.yaml"), []byte(content)))
+	return fSys, tmpDir
+}
+
+func checkYaml(actual resmap.ResMap, expected string, assert *assert.Assertions) {
+	yml, err := actual.AsYaml()
+	assert.NoError(err)
+	assert.Equal(expected, string(yml))
+}
+
+func testRemoteResource(assert *assert.Assertions, kustomization string, expected string) {
+	fSys, tmpDir := createKusDir(kustomization, assert)
+
+	b := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
 	m, err := b.Run(
 		fSys,
 		tmpDir.String())
-	if utils.IsErrTimeout(err) {
-		// Don't fail on timeouts.
-		t.SkipNow()
+
+	assert.NoError(err)
+	checkYaml(m, expected, assert)
+
+	assert.NoError(fSys.RemoveAll(tmpDir.String()))
+}
+
+func TestRemoteLoad(t *testing.T) {
+	assert := assert.New(t)
+
+	fSys := filesys.MakeFsOnDisk()
+	b := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
+
+	m, err := b.Run(
+		fSys,
+		"github.com/kubernetes-sigs/kustomize/examples/multibases/dev/?ref=v1.0.6")
+	assert.NoError(err)
+	checkYaml(m, multibaseDevExampleBuild, assert)
+}
+
+func TestRemoteResourceHttps(t *testing.T) {
+	tests := map[string]remoteResourceCase{
+		"basic": {
+			kustomization: `
+resources:
+- https://github.com/kubernetes-sigs/kustomize//examples/multibases/dev/?ref=v1.0.6`,
+		},
+		".git repo suffix, no slash suffix": {
+			kustomization: `
+resources:
+- https://github.com/kubernetes-sigs/kustomize.git//examples/multibases/dev?ref=v1.0.6`,
+		},
+		"repo": {
+			kustomization: `
+resources:
+- https://github.com/annasong20/kustomize-test.git?ref=main`,
+		},
+		"raw remote file": {
+			kustomization: `
+resources:
+- https://raw.githubusercontent.com/kubernetes-sigs/kustomize/v3.1.0/examples/multibases/base/pod.yaml
+namePrefix: dev-`,
+		},
+	}
+
+	for name, test := range tests {
+		test := test
+		if !test.skip {
+			t.Run(name, func(t *testing.T) {
```

Comment:
We should reverse the logic:

```
		test := test
		t.Run(name, func(t *testing.T) {
			if test.skip {
                 t.SkipNow()
			}
            testRemoteResource...
```

The reason for using t.SkipNow() when applicable is that it will explicitly let you know that a test has been skipped. Otherwise it's like the test isn't even there and we may forget about it. 

E.g. with a kpt example:

```
=== RUN   Test_GitParseArgs
--- PASS: Test_GitParseArgs (2.16s)
=== RUN   Test_GitParseArgs/http_url_with_.git_suffix
    --- PASS: Test_GitParseArgs/http_url_with_.git_suffix (1.27s)
=== RUN   Test_GitParseArgs/starts_with_github.com
    --- PASS: Test_GitParseArgs/starts_with_github.com (0.89s)
PASS
```

vs 

```
=== RUN   Test_GitParseArgs
--- PASS: Test_GitParseArgs (2.88s)
=== RUN   Test_GitParseArgs/git@_url_with_.git_suffix
    --- SKIP: Test_GitParseArgs/git@_url_with_.git_suffix (0.00s)

Test ignored.
=== RUN   Test_GitParseArgs/http_url_with_.git_suffix
    --- PASS: Test_GitParseArgs/http_url_with_.git_suffix (2.00s)
=== RUN   Test_GitParseArgs/starts_with_github.com
    --- PASS: Test_GitParseArgs/starts_with_github.com (0.88s)

Test ignored.
PASS
```

The latter is more informative and desirable. 

Same comment applies in all other tests.